### PR TITLE
Add `Cursor.describe` to retrieve the schema of a query

### PR DESCRIPTION
**Pull request by mdesmet**
## Release notes

( ) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: trino/dbapi.py [hunk 1]
```diff
@@ -220,6 +220,20 @@ def cursor(self, experimental_python_types: bool = None):
         )
 
 
+class DescribeOutput(NamedTuple):
```
[the remark this answers is not in the document]
Comment (mdesmet) [reply trino/dbapi.py:223]: In this case we only want an immutable structure with attributes, which is exactly the use case for a `NamedTuple`.

File: tests/integration/test_dbapi_integration.py [hunk 2]
```diff
@@ -1107,3 +1108,69 @@ def test_prepared_statements(run_trino):
...
+            type_size=8,
+            aliased=False,
+        ),
+        DescribeOutput(
+            name='comment',
+            catalog='tpch',
+            schema='tiny',
+            table='nation',
+            type='varchar(152)',
+            type_size=0,
```
[the remark this answers is not in the document]
Comment (mdesmet) [reply tests/integration/test_dbapi_integration.py:1173]: I'm not exactly sure it is returned by `DESCRIBE OUTPUT`

```
❯ trino
trino> prepare test FROM select * from tpch.tiny.customer;
PREPARE
trino> describe output test;
 Column Name | Catalog | Schema |  Table   |     Type     | Type Size | Aliased
-------------+---------+--------+----------+--------------+-----------+---------
 custkey     | tpch    | tiny   | customer | bigint       |         8 | false
 name        | tpch    | tiny   | customer | varchar(25)  |         0 | false
 address     | tpch    | tiny   | customer | varchar(40)  |         0 | false
 nationkey   | tpch    | tiny   | customer | bigint       |         8 | false
 phone       | tpch    | tiny   | customer | varchar(15)  |         0 | false
 acctbal     | tpch    | tiny   | customer | double       |         8 | false
 mktsegment  | tpch    | tiny   | customer | varchar(10)  |         0 | false
 comment     | tpch    | tiny   | customer | varchar(117) |         0 | false
(8 rows)

Query 20221214_180638_00003_8aypw, FINISHED, 1 node
Splits: 1 total, 1 done (100,00%)
0,72 [0 rows, 0B] [0 rows/s, 0B/s]
```

File: trino/dbapi.py [hunk 3]
```diff
@@ -517,6 +531,27 @@ def fetchmany(self, size=None) -> List[List[Any]]:
 
         return result
 
+    def describe(self, sql: str) -> List[DescribeOutput]:
+        """
+        Returns the schema of a given SQL statement
```
Comment (hovaesco) [inline trino/dbapi.py:536]: It returns not only schema, please rephrase it.

**Review by hovaesco — COMMENTED** [review]
Looks good to me.

File: trino/dbapi.py (hunk 3 above)
Comment (mdesmet) [reply trino/dbapi.py]: Done

**Review by hovaesco — APPROVED** [review]

**Review by hashhar — APPROVED** [review]
LGTM

Looks useful. Similar to `ResultSetMetaData` from JDBC land.

**Pull request merged**